### PR TITLE
Service Catalog support fixes and tests

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: integration_tests/main_test.go [hunk 1]
```diff
@@ -38,16 +37,16 @@ func TestWorkflow(t *testing.T) {
 			Resources: bundleResources(t),
 		},
 	}
-	setupApp(t, bundle, false, testWorkflow)
+	setupApp(t, bundle, false, true, testWorkflow)
```
Comment (nilebox) [inline integration_tests/main_test.go]: two boolean flags are confusing, I would prefer passing a config struct with fields

File: integration_tests/service_catalog_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		},
+		Spec: smith.BundleSpec{
+			Resources: []smith.Resource{
+				{
+					Name: smith.ResourceName(instance.Name),
+					Spec: toUnstructured(t, instance),
+				},
+				{
+					Name:      smith.ResourceName(binding.Name),
+					DependsOn: []smith.ResourceName{smith.ResourceName(instance.Name)},
```
Comment (nilebox) [inline integration_tests/service_catalog_test.go]: IIRC you can omit smith.ResourceName in the array declaration
```
DependsOn: []{smith.ResourceName(instance.Name)},
```

File: integration_tests/service_catalog_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+					Spec:      toUnstructured(t, binding),
+				},
+			},
+		},
+	}
+	setupApp(t, bundle, true, true, testServiceCatalog)
+}
+
+func testServiceCatalog(t *testing.T, ctx context.Context, namespace string, bundle *smith.Bundle, config *rest.Config, clientset *kubernetes.Clientset,
```
Comment (nilebox) [inline integration_tests/service_catalog_test.go]: 10 input arguments + optional args... Is this a good practice in Go?

Comment (ash2k) [inline integration_tests/service_catalog_test.go]: It is not great, but this is just a test. I'll see if it would be better to put it all in a struct.

**Pull request merged**